src/d3a/models/market/one_sided.py

```diff
@@ -117,9 +118,11 @@ def offer(self, price: float, energy: float, seller: str, seller_origin,
 
         if offer_id is None:
             offer_id = self.bc_interface.create_new_offer(energy, price, seller)
-        offer = Offer(offer_id, self.now, price, energy, seller, original_price,
+        now = self.now if self.time_slot else time_slot
```

Comment:
I have the feeling that we actually want the opposite here, right?
If this is true, this is relevant to the other usages below
```suggestion
        now = time_slot if time_slot else self.now
```

Another nit, it would be great if we rename `now` to `time`